### DISCUSSION TITLE
xsbug: how to reset the target?

### DISCUSSION THREAD

**Discussion by tve** [Moddable-OpenSource/moddable#992] [Getting started] [answered]
If I write a trivial main that consists of just `trace("Hello World!\n")` xsbug shows the message in the log and then everything seems pretty dead. How can I soft-restart the app/system? I also don't see any way to hard-reset, i.e. toggle the reset line (e.g. via DTR/RST for most esp32 boards)?

**Comment by andycarle** [comment, marked as answer]
> How can I soft-restart the app/system?

Hit the stop button in xsbug.

![image](https://user-images.githubusercontent.com/1427817/207821597-ec0f1e5b-db0f-454c-b363-26085d3a3c3c.png)

>  I also don't see any way to hard-reset, i.e. toggle the reset line (e.g. via DTR/RST for most esp32 boards)?

True. Most of the boards that we work with routinely have a physical reset button that isn't terribly painful to get to. We do have a lot of experience manipulating the DTR/RST lines on the ESP32 from host machines, but I don't believe that is exposed in xsbug.

**Reply by tve** [reply]
> Hit the stop button in xsbug.

If you can see it 😏 ... when the program stops, it's not there. You have to know to click on the main tab to make it appear. Or perhaps that's only the case if you have the serial tab as first tab?
Aside: that icon says "stop" to me when it really is "restart".

**Reply by andycarle** [reply]
It really is "stop," which you'll find out if you try it when debugging a target running in the simulator. It just happens that most (maybe all?) of our microcontroller targets are set up to restart if nothing is running.
